File: Back to Basics Approach 3rd Edition/cse_143/cs2_sections/section01_arrayList/removeDuplicates.java
package cse_143.cs2_sections.section01_arrayList;

import java.util.ArrayList;
import java.util.List;

/**
 * https://practiceit.cs.washington.edu/problem/view/cs2/sections/arraylist/removeDuplicates
 * <p>
 * Takes as a parameter a sorted ArrayList of Strings and that
 * eliminates any duplicates from the list.
 *
 * @author Alex Golub
 * @since 21-Apr-21, 10:25 PM
 */
class removeDuplicates {
  public static void removeDuplicates(List<String> list) {
    if (list == null || list.size() == 0) {
      return;
    }
    
    List<String> noDup = new ArrayList<>();
    noDup.add(list.get(0));
    for (int i = 1; i < list.size(); i++) {
      if (!list.get(i).equals(list.get(i - 1))) {
        noDup.add(list.get(i));
      }
    }
    list.clear();
    list.addAll(noDup);
  }
}

File: Back to Basics Approach 3rd Edition/cse_143/cs2_sections/section01_arrayList/removeDuplicates.js
/**
 * @author Alex Golub
 * @since 21-Apr-21, 10:29 PM
 */
// const list = ["be", "be", "is", "not", "or", "question", "that", "the", "to", "to"];
const list = ['duplicate', 'duplicate', 'duplicate', 'duplicate', 'duplicate'];
const noDup = [list[0]]; // pre: list.length > 0
for (let i = 1; i < list.length; i++) {
  if (list[i] !== list[i - 1]) {
    noDup.push(list[i]);
  }
}

list.length = 0; // remove all content from list
for (const word of noDup) {
  list.push(word);
}
console.log(list);
